crates/compiler/solve/src/solve.rs

```diff
@@ -1771,6 +1771,57 @@ fn solve(
 
                 state
             }
+            IngestedFile(type_index, _, bytes) => {
+                let actual = either_type_index_to_var(
+                    subs,
+                    rank,
+                    pools,
+                    problems,
+                    abilities_store,
+                    obligation_cache,
+                    &mut can_types,
+                    aliases,
+                    *type_index,
+                );
+
+                if let Success { .. } = unify(
+                    // TODO: if we don't clone and a later branch matches, we will get a failure in alias analysis.
+                    // That said, I assume cloning is expensive and should be avoided.
+                    &mut UEnv::new(&mut subs.clone()),
+                    actual,
+                    Variable::LIST_U8,
+                    Mode::EQ,
+                    Polarity::OF_VALUE,
+                ) {
+                    // List U8 always valid.
+                    state
+                } else if let Success { .. } = unify(
+                    &mut UEnv::new(&mut subs.clone()),
+                    actual,
+                    Variable::STR,
+                    Mode::EQ,
+                    Polarity::OF_VALUE,
+                ) {
+                    // Str only valid if valid utf8.
+                    if std::str::from_utf8(bytes).is_err() {
```

Comment:
Digging into the mono conversion, I think this could be deferred if we create it as a layout problem instead. That would avoid adding the bytes to the constraints. Then this would just be type checking and the mono conversion would check anything else (bad utf8, decode error, etc). Does that sound like something we would want?

Reply:
I don't think we want to have `roc check` ever be forced to enter mono - we did some work a while back to remove that dependency. I think we need to check all of this here.